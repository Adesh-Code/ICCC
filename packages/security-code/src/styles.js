const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
