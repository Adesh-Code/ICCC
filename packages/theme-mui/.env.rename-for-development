# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE