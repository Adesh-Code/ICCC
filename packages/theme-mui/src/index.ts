// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
