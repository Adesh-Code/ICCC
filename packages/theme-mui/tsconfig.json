{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
