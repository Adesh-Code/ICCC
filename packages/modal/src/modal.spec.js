import React from "react";
import Button from "@mui/material/Button";
import Modal from "./modal";
import { fireEvent, render, screen, within } from "@testing-library/react";
import Typography from "@iccc/typography";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const onChangeMock = jest.fn();

const arrange = (options) => {
  const args = {
    actions: <Button>Modal action</Button>,
    content: "Modal body",
    open: true,
    title: "Modal title",
    ...options
  };

  return render(
    <ThemeProvider theme={light}>
      <Modal {...args} />
    </ThemeProvider>
  );
};

describe("Modal component", () => {
  it("renders a modal", () => {
    arrange();
    const modal = screen.getByRole("dialog");
    expect(modal).toMatchSnapshot();
  });

  it("renders with title, content, actions, and close button", () => {
    arrange();
    const modal = screen.getByRole("dialog");
    const closeButton = screen.getByLabelText("close");

    expect(modal).toHaveTextContent("Modal title");
    expect(modal).toHaveTextContent("Modal body");
    expect(modal).toHaveTextContent("Modal action");
    expect(closeButton).toBeInTheDocument();
  });

  it("doesn't render actions if none are provided", () => {
    arrange({ actions: null });
    expect(screen.queryByText("Modal action")).not.toBeInTheDocument();
  });

  it("renders subdivision style variant", () => {
    arrange({ styleVariant: "subdivision" });
    const modal = screen.getByRole("dialog");
    expect(modal).toHaveClass("DhigModal--styleVariant--subdivision");
  });

  it("renders lightbox type", () => {
    arrange({ type: "lightbox" });
    const modal = screen.getByRole("dialog");
    const closeButton = screen.getByLabelText("close");

    expect(modal.firstChild).not.toHaveClass("MuiDialogTitle-root");
    expect(closeButton).toHaveClass("DhigModal--closeButton--outside");
  });

  it("renders lightbox type with caption content", () => {
    const caption = (
      <>
        <Typography variant="caption">
          This is a placeholder caption.
        </Typography>
        <Typography variant="smallprint">
          Image credit: © 2022 Faucibus imperdiet integer leo nunc urna et
          gravida lobortis quam.
        </Typography>
      </>
    );
    arrange({ type: "lightbox", caption });
    const modal = screen.getByRole("dialog");
    const closeButton = screen.getByLabelText("close");
    const captionText = screen.getByText("This is a placeholder caption.");
    const creditText = screen.getByText(
      "Image credit: © 2022 Faucibus imperdiet integer leo nunc urna et gravida lobortis quam."
    );

    expect(modal.firstChild).not.toHaveClass("MuiDialogTitle-root");
    expect(captionText).toHaveClass("MuiTypography-caption");
    expect(creditText).toHaveClass("MuiTypography-smallprint");
    expect(closeButton).toHaveClass("DhigModal--closeButton--outside");
  });

  it("renders lightbox type with a string in the caption prop", () => {
    arrange({ type: "lightbox", caption: "This is a placeholder caption." });
    const modal = screen.getByRole("dialog");
    const closeButton = screen.getByLabelText("close");
    const captionText = screen.getByText("This is a placeholder caption.");

    expect(modal.firstChild).not.toHaveClass("MuiDialogTitle-root");
    expect(captionText).toHaveClass("MuiTypography-caption");
    expect(closeButton).toHaveClass("DhigModal--closeButton--outside");
  });

  it("renders alert type", () => {
    // eslint-disable-next-line react/display-name
    arrange({ type: "alert", AlertIcon: () => <div /> });
    const modal = screen.getByRole("dialog");

    expect(modal).toHaveClass("DhigModal--type--alert");
  });

  it("renders alert type without AlertIcon", () => {
    arrange({ type: "alert" });
    const modal = screen.getByRole("dialog");

    expect(modal).toHaveClass("DhigModal--type--alert");
  });

  it("renders different max-widths", () => {
    arrange({ maxWidth: "xl" });
    const modal = screen.getByRole("dialog");
    expect(modal).toHaveClass("MuiDialog-paperWidthXl");
  });

  it("renders with logo", () => {
    arrange({ displayLogo: true });
    const modal = screen.getByRole("dialog");
    const logo = screen.getByAltText("Autodesk logo");

    expect(modal).not.toHaveTextContent("Modal title");
    expect(logo).toBeInTheDocument();
  });

  it("renders without close button", () => {
    arrange({ closeButtonDisplay: false });
    const closeButton = screen.queryByLabelText("close");
    expect(closeButton).not.toBeInTheDocument();
  });

  it("renders without a title", () => {
    const { container } = arrange({ title: null });

    const h2 = container.querySelector("h2");
    expect(h2).toBeNull();
  });

  it("renders with accessibility props", () => {
    arrange({
      "aria-describedby": "aria-described-by-test",
      "aria-labelledby": "aria-labelled-by-test"
    });
    const modal = screen.getByRole("dialog");

    expect(modal).toHaveAttribute("aria-labelledby", "aria-labelled-by-test");
    expect(modal).toHaveAttribute("aria-describedby", "aria-described-by-test");
  });

  it("accepts close button props", () => {
    arrange({
      CloseButtonProps: {
        accessibleLabel: "test-label",
        handleClose: () => {},
        onClick: onChangeMock
      }
    });
    const modal = screen.getByRole("dialog");
    const closeButton = within(modal).getAllByRole("button")[0];
    expect(closeButton).toHaveAttribute("aria-label", "test-label");

    fireEvent.click(closeButton);
    expect(onChangeMock).toBeCalledTimes(1);
  });
});
