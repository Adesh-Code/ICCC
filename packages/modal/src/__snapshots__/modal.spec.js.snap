// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders a modal 1`] = `
<div
  aria-labelledby="mui-1"
  class="MuiPaper-root DhigModal--styleVariant--default DhigModal--type--modal MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm emotion-0"
  role="dialog"
>
  <button
    aria-label="close"
    class="MuiButtonBase-root MuiIconButton-root DhigModal--closeButton--default MuiIconButton-sizeMedium emotion-1"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="utility/cross"
        >
          <path
            d="M19 5L12 12M12 12L5 19M12 12L5 5M12 12L19 19"
            id="stroke"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </g>
      </svg>
    </svg>
  </button>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root emotion-3"
    id="mui-1"
  >
    Modal title
  </h2>
  <div
    class="MuiDialogContent-root emotion-4"
  >
    Modal body
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing emotion-5"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium emotion-6"
      tabindex="0"
      type="button"
    >
      Modal action
    </button>
  </div>
</div>
`;
